agent/connectionset/connection_set.go

```diff
@@ -0,0 +1,159 @@
+// pmm-agent
+// Copyright 2019 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connectionset
+
+import (
+	"context"
+	"sync"
+	"time"
+)
+
+const periodForRunningDeletingOldEvents = time.Minute
+
+type ConnectionSet struct {
```

Comment:
I think we need to redisign this.
It is better to designed it as a service, responsible for uptime calculation. It should provide a simple API.

Reply:
Agree,

At the start I thought that it will be a struct just to contain a connection events, but then it became a service.
Renamed it.